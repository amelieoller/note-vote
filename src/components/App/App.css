.container {
	text-align: center;
	font-family: 'Roboto', sans-serif;
	color: #212121;
}
