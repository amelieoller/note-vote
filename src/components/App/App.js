import React, { Component } from 'react';
import styles from './App.css';
import NoteCard from '../NoteCard/NoteCard';
import _ from 'lodash';
import NoteForm from '../NoteForm/NoteForm';
import { database } from '../../firebase';

class App extends Component {
	constructor(props) {
		super(props);
		this.state = {
			title: '',
			body: '',
			notes: [
				{
					title: 'First Note Title',
					body: 'First Note Body'
				},
				{
					title: 'Second Note Title',
					body: 'Second Note Body'
				}
			]
		};

		// bind
		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleChange(e) {
		this.setState({
			[e.target.name]: e.target.value
		});
	}

	handleSubmit(e) {
		e.preventDefault();
		const note = {
			title: this.state.title,
			body: this.state.body
		};
		database.push(note);
		this.setState({
			title: '',
			body: ''
		})
	}

	renderNotes() {
		return _.map(this.state.notes, (note, key) => {
			return (
				<NoteCard key={key}>
					<h3>{note.title}</h3>
					<p>{note.body}</p>
				</NoteCard>
			);
		});
	}

	render() {
		return (
			<div className={styles.container}>
				<h1>Note Vote</h1>
				<NoteForm
					handleChange={this.handleChange}
					handleSubmit={this.handleSubmit}
					state={this.state}
				/>
				{this.renderNotes()}
			</div>
		);
	}
}

export default App;
