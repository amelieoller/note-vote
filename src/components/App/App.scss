@import '../../shared/colors.scss';
@import '../../shared/mixins.scss';

.container {
	display: grid;
	grid-template-columns: repeat(12, 1fr);
	grid-template-rows: 50px auto 50px;
	min-height: 100vh;
}

main {
	grid-column: span 10;
}

header {
	grid-column: span 2;
	color: $white;
	background-color: $red;
}

.logoType {
	font-size: 1.1em;
	position: relative;
	float: left;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
}

footer {
	grid-column: span 12;
	background-color: $red;
	color: $white;
	display: flex;
	align-items: center;
	justify-content: center;
}

.searchContainer {
	overflow: hidden;
}

.searchInput {
	float: right;
}
