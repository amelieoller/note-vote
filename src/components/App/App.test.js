import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount, render } from 'enzyme';

describe('App Component', () => {
	test('renders without crashing', () => {
		const wrapper = shallow(<App>Note Vote</App>);
		expect(wrapper).toMatchSnapshot();
	});

});
