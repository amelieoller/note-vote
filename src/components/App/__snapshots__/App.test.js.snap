// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
<div
  className="container"
>
  <h1>
    Note Vote
  </h1>
</div>
`;
